src/js/stepped-progress.js

```diff
@@ -1,24 +1,172 @@
+ *
+ * @access private
+ * @type {Object}
+ */
+const classNames = {
+	step: 'o-stepped-progress__step'
+};
+
+/**
+ * Represents a stepped progress component.
+ */
 class SteppedProgress {
 
 	/**
 	 * Class constructor.
-	 * @param {HTMLElement} [steppedProgressElement] - The component element in the DOM
-	 * @param {Object} [options={}] - An options object for configuring the component
+	 *
+	 * @access public
+	 * @param {HTMLElement} steppedProgressElement - The component element in the DOM.
+	 * @param {Object} [options={}] - An options object for configuring the component.
 	 */
-	constructor (steppedProgressElement, opts) {
+	constructor (steppedProgressElement, options) {
 		this.steppedProgressElement = steppedProgressElement;
 		this.options = Object.assign({}, {
 			// TODO
-		}, opts || SteppedProgress.getDataAttributes(steppedProgressElement));
+		}, options || SteppedProgress.getDataAttributes(steppedProgressElement));
+		this._constructSteps();
+	}
+
+	/**
+	 * Get an array of steps.
+	 *
+	 * @access public
+	 * @returns {Array<SteppedProgressStep>} Returns an array of steps.
+	 */
+	getSteps() {
+		return [...this._steps];
+	}
+
+	/**
+	 * Get an array of steps with a "completed" status.
+	 *
+	 * @access public
+	 * @returns {Array<SteppedProgressStep>} Returns an array of steps.
+	 */
+	getCompletedSteps() {
+		return this._steps.filter(step => step.isComplete());
+	}
+
+	/**
+	 * Get whether a step exists at a given index (0-based).
+	 *
+	 * @access public
+	 * @param {Number} index - The index to check.
+	 * @returns {Boolean} Returns whether a step exists at a given index.
+	 */
+	hasStepAtIndex(index) {
+		return Boolean(this._steps[index]);
+	}
+
+	/**
+	 * Get the step at a given index (0-based).
+	 *
+	 * @access public
+	 * @param {Number} index - The index of the step to get.
+	 * @returns {SteppedProgressStep} Returns the step at the given index.
+	 * @throws {Error} Will throw an error if there is no step at the given index. Use {@link SteppedProgress#hasStepAtIndex} to check.
+	 */
+	getStepAtIndex(index) {
+		if (!this.hasStepAtIndex(index)) {
+			throw new Error(`No step at index: ${index}`);
+		}
+		return this._steps[index];
+	}
+
+	/**
+	 * Get the step which has the "current" state. If there are multiple steps with this state then
+	 * the last one will be returned.
+	 *
+	 * @access public
+	 * @returns {SteppedProgressStep} Returns the current step.
+	 */
+	getCurrentStep() {
+		return this._steps.filter(step => step.isCurrent()).pop();
+	}
+
+	/**
+	 * Get the last step in the stepped progress.
+	 *
+	 * @access public
+	 * @returns {SteppedProgressStep} Returns the last step.
+	 */
+	getLastStep() {
+		return this._steps[this._steps.length - 1];
+	}
+
+	/**
+	 * Get whether all steps have the "completed" state.
+	 *
+	 * @access public
+	 * @returns {Boolean} Returns whether all steps are completed.
+	 */
+	isComplete() {
+		return (this.getCompletedSteps().length === this._steps.length);
```

Comment:
I'd be tempted to use Array.prototype.every instead of using Array.prototype.filter and comparing lengths. The reasons I would use every are: I think it is more readable and simpler to understand as it describes exactly what the intention is; It is faster because it would return `false` as soon as it finds an incomplete step.
```suggestion
		return this._steps.every(step => step.isComplete())
```